pkg/storage/snapshot/restore.go

```diff
@@ -98,6 +98,87 @@ func restoreDVName(vmRestore *snapshotv1.VirtualMachineRestore, name string) str
 	return restorePVCName(vmRestore, name)
 }
 
+func updatePVCOwner(vr *snapshotv1.VolumeRestore, pvc *corev1.PersistentVolumeClaim, t *vmRestoreTarget) (bool, error) {
+	if vr.DataVolumeName == nil {
```

Comment:
Can you please flip the condition and return earlier. In this way, we can avoid an additional nesting level